CI-API.Data/Repository/AdminRepository.cs

```diff
@@ -330,5 +236,529 @@ private JsonResult updateAddUser(User userHasTobeUpdated,UserDetailViewModel use
+                }
+                else
+                {
+                    return new JsonResult(new apiResponse<string> { Message = ResponseMessages.MissionNotUpdatedSuccessfully, StatusCode = responseStatusCode.RequestFailed, Result = false });
+                }
+            }
+            catch
+            {
+                return new JsonResult(new apiResponse<string> { Message = ResponseMessages.InternalServerError, StatusCode = responseStatusCode.BadRequest, Result = false });
+            }
+        }
+        #endregion
+
+        #region DeleteMission
+        public async Task<JsonResult> deleteMission(long? missionId)
+        {
+            try
+            {
+                Mission? mission = await Task.FromResult(cIDbContext.Missions.Where(M => M.MissionId == missionId).FirstOrDefault());
+                if (mission != null)
+                {
+                    mission.DeletedAt = DateTime.Now;
+                    mission.Status = StaticCode.MissionInactive;
+
+                    GoalMission? goalMission = await Task.FromResult(cIDbContext.GoalMissions.Where(G => G.MissionId == missionId).FirstOrDefault());
+                    if (goalMission != null)
+                    {
+                        goalMission.DeletedAt = DateTime.Now;
+                        cIDbContext.SaveChanges();
+                    }
+
+                    List<MissionMedium>? missionOldMedia = await Task.FromResult(cIDbContext.MissionMedia.Where(MM => MM.MissionId == missionId).ToList());
+                    if (missionOldMedia != null)
+                    {
+                        foreach (var media in missionOldMedia)
+                        {
+                            media.DeletedAt = DateTime.Now;
+                            cIDbContext.SaveChanges();
+                        }
+                    }
+
+                    List<MissionDocument>? missionOldDocuments = await Task.FromResult(cIDbContext.MissionDocuments.Where(MD => MD.MissionId == missionId).ToList());
+                    if (missionOldDocuments != null)
+                    {
+                        foreach (var document in missionOldDocuments)
+                        {
+                            document.DeletedAt = DateTime.Now;
+                            cIDbContext.SaveChanges();
+                        }
+                    }
+
+                    List<MissionSkill>? missionSkill = await Task.FromResult(cIDbContext.MissionSkills.Where(MS => MS.MissionId == missionId).ToList());
+                    if (missionSkill != null)
+                    {
+                        foreach (var skill in missionSkill)
+                        {
+                            skill.DeletedAt = DateTime.Now;
+                            cIDbContext.SaveChanges();
+                        }
+                    }
+
+                   
```

Comment:
apply cIDbContext.SaveChanges() after all above crud operation for deleting mission

Reply:
changes done